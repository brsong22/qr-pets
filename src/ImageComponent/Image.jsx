import React from 'react';
import PropTypes from 'prop-types';

const Image = ({ src, alt }) => (
  <img src={require(`${src}`)} alt={alt} />
);

Image.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
};

Image.defaultProps = {
  src: '',
  alt: '',
};

export default Image;
