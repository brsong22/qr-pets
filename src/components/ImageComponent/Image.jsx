import React from 'react';
import PropTypes from 'prop-types';

const Image = ({
  imageClassName, src, alt, w, h,
}) => (
  <img className={imageClassName} src={src} alt={alt} width={w} height={h} />
);

Image.propTypes = {
  imageClassName: PropTypes.string,
  src: PropTypes.string,
  alt: PropTypes.string,
  w: PropTypes.string,
  h: PropTypes.string,
};

Image.defaultProps = {
  imageClassName: '',
  src: '',
  alt: '',
  w: '100',
  h: '100',
};

export default Image;
