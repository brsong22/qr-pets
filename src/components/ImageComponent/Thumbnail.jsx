import React from 'react';
import PropTypes from 'prop-types';
import Image from './Image';

const Thumbnail = ({
  src, alt, w, h,
}) => (
  <Image src={src} alt={alt} width={w} height={h} />
);

Thumbnail.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  w: PropTypes.string,
  h: PropTypes.string,
};

Thumbnail.defaultProps = {
  src: '',
  alt: '',
  w: '100',
  h: '100',
};

export default Thumbnail;
