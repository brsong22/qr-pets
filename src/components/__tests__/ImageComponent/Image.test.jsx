
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ImageComponent from '../../ImageComponent/Image';

describe('ImageComponent', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<ImageComponent />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders props as expected', () => {
    const props = {
      src: 'https://s3.us-east-2.amazonaws.com/qr-pets-images/IMG_3855.JPG',
      alt: 'test',
      w: 10,
      h: 10,
    };
    const wrapper = shallow(
      <ImageComponent
        src={props.src}
        alt={props.alt}
        width={props.w}
        height={props.h}
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
