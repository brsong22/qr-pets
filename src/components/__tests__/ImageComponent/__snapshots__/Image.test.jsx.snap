// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders as expected 1`] = `
<img
  alt=""
  className=""
  height="100"
  src=""
  width="100"
/>
`;

exports[`Image renders props as expected 1`] = `
<img
  alt="test"
  className=""
  height="100"
  src="https://s3.us-east-2.amazonaws.com/qr-pets-images/IMG_3855.JPG"
  width="100"
/>
`;
